lz4.go

```diff
@@ -122,18 +142,29 @@ func (w *Writer) Write(src []byte) (int, error) {
 		return 0, err
 	}
 
-	w.inpBufIndex = (w.inpBufIndex + 1) % 2
 	w.totalCompressedWritten += written + 4
 	return len(src), nil
 }
 
+func (w *Writer) nextInputBuffer() []byte {
+	w.inpBufIndex = (w.inpBufIndex + 1) % 2
+	tmpSlice := reflect.SliceHeader{
```

Comment:
rather than use reflection, the cgo helper function C.GoBytes can be used to convert the underlying C buffer to a slice. The Read function does this, see cgo docs: https://golang.org/cmd/cgo/#hdr-Go_references_to_C

Reply:
I removed that because calling this function creates an extra copy of the data.

> A few special functions convert between Go and C types by making copies of the data

Looking at Go issues, it seems using `SliceHeader` is the only sane way to create a slice from a memory pointer without copying it.